src/lib/index/subscribe.js

```diff
@@ -35,31 +36,35 @@ const
 
 	emitter = new EventEmitter(),
 
-	subscribeAction = ({ topic, handler, channel }) => {
+	subscribeAction = ({ topic, handler, channel, config }) => {
 		// Ensure the topic exists
 		channel.assertQueue(topic);
-
+		// Set prefetch
+		if (config.prefetch && _.isInteger(config.prefetch)) {
+			channel.prefetch(config.prefetch);
+		}
 		// Subscribe to the topic
 		return channel.consume(topic, message => {
 			return Promise.resolve()
 				.then(() => {
-					// Invoke the handler with the message
+					// Invoke the handler with the message, and return the result promise or a resolution
 					try {
 						const handlerResult = handler(message.content);
-						Promise.resolve(handlerResult).catch(err => {
+						return Promise.resolve(handlerResult).catch(err => {
 							emitter.emit(ERROR_EVENT, err.message);
```

Comment:
Unnecessary duplication of catch conditions. Isn't it just...
```javascript
return Promise.resolve(message.content)
    .then(handler)
    .catch(err => emitter.emit(ERROR_EVENT, err.message));
```

Reply:
This one is actually worth doing, ill do it.